src/mixins/itext_behavior.mixin.js

```diff
@@ -412,6 +418,253 @@
+        this.setCursorByClick(options.e);
+        this._updateTextarea();
+        this.restartCursorIfNeeded();
+        this.renderCursorOrSelection();
+      }
+    },
+
+    /**
+     * support native like text dragging
+     * @private
+     */
+    dragLeaveHandler: function () {
+      if (this.__isDraggingOver) {
+        this.__isDraggingOver = false;
+        this.exitEditing();
+      }
+    },
+
+    /**
+     * support native like text dragging
+     * fired only on the drag source
+     * https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API/Drag_operations#finishing_a_drag
+     * @private
+     * @param {DragEvent} e
+     * @returns {boolean} should handle event
+     */
+    onDragEnd: function (e) {
+      if (this.__isDragging && this.__dragStartFired) {
+        if (this.__dragStartSelection) {
+          var selectionStart = this.__dragStartSelection.selectionStart;
+          var selectionEnd = this.__dragStartSelection.selectionEnd;
+          var trailingSpace = this.__dragStartSelection.trailingSpace;
+          if (e.dataTransfer.dropEffect === 'move') {
+            this.insertChars('', null, selectionStart, selectionEnd);
+            this.selectionStart = this.selectionEnd = selectionStart;
+            this.hiddenTextarea && (this.hiddenTextarea.value = this.text);
+            this._updateTextarea();
+            this.fire('changed', { index: selectionStart, action: 'dragend' });
+            this.canvas.fire('text:changed', { target: this });
+            this.canvas.requestRenderAll();
+          }
+          else {
+            this.selectionStart = selectionStart;
+            this.selectionEnd = selectionEnd - trailingSpace;
+          }
+          this.exitEditing();
+          this.__lastSelected = false;
+        }
+
+        this.fire('dragend', { e: e });
+      }
+
+      this.__dragImageDisposer && this.__dragImageDisposer();
+      delete this.__dragImageDisposer;
+      delete this.__dragStartSelection;
+      this.__isDraggingOver = false;
+    },
+
+    /**
+     * support native like text dragging
+     *
+     * Override the `text/plain | application/fabric` types of {@link DragEvent#dataTransfer}
+     * in order to change the drop value or to customize styling respectively, by listening to the `drop:before` event
+     * https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API/Drag_operations#performing_a_drop
+     * @private
+     */
+    dropHandler: function (options) {
+      var e = options.e;
+      this.__isDraggingOver = false;
+      // inform browser that the drop has been accepted
+      e.preventDefault();
+      var insert = e.dataTransfer.getData('text/plain');
+      if (insert) {
+        var insertAt = this.selectionStart;
+        var data = e.dataTransfer.types.includes('application/fabric') ?
+          JSON.parse(e.dataTransfer.getData('application/fabric')) :
+          {};
+        var styles = data.styles;
```

Comment:
@ShaMan123 @asturur 

Something to consider here. This currently allows dragging of inline styles from one object to another. Some applications enforce a one style per object behavior restriction, using the `fabric.disableStyleCopyPaste` property to prevent styles from being copied from one object to another. I'd be inclined to put a check for this property here as well. Maybe rename it to `useDestinationStyle`?

Reply:
hmmm I don't like the flag itself because it isn't flexible enough.
You mean to have a property on Text? It's so diverse, that's why I left methods public to make it easy to override.
You simply listen to the `drop:before` event and use `DragEvent.setData('application/fabric',...)` to change the data accordingly, Then the `drop` handler` won't have style data to work with, or whatever the dev does.
I think this is best practice, it's how most frameworks do it.

Reply:
Or even from `dragstart` using `effectAllowed`, many possibilities